concent_api/core/views.py

```diff
@@ -346,21 +481,50 @@ def store_message_and_message_status(golem_message_type, task_id, raw_golem_mess
     golem_message.full_clean()
     golem_message.save()
 
-    if status == ReceiveStatus:
-        receive_message_status  = ReceiveStatus(
+    if status is not None:
+        receive_message_status  = status(
             message     = golem_message,
-            timestamp   = message_timestamp
+            timestamp   = message_timestamp,
+            delivered   = delivered
         )
         receive_message_status.full_clean()
         receive_message_status.save()
 
-    elif status == ReceiveOutOfBandStatus:
-        receive_out_of_band_status = ReceiveOutOfBandStatus(
-            message     = golem_message,
-            timestamp   = message_timestamp,
-            delivered   = True
-        )
-        receive_out_of_band_status.full_clean()
-        receive_out_of_band_status.save()
+
+def get_file_status(file_transfer_token_from_database: message.concents.FileTransferToken) -> bool:
+    slash = '/'
+    assert file_transfer_token_from_database.files[0]['path'][0]    is not slash
+    assert settings.STORAGE_CLUSTER_ADDRESS[-1]                     is  slash
+    assert len(file_transfer_token_from_database.files)             == 1
```

Comment:
This assert should be at the top. If there are zero files the code will not even get here.